src/LLVMSupport/SMTConstraintGenerator.cpp

```diff
@@ -166,6 +167,24 @@ class SMTConstraintGenerator::Internals {
     g->symbolicPlacements.insert(make_pair(i, placement));
   }
 
+  static void constrainMemoryInstruction(SMTConstraintGenerator *g, Instruction *i,
+    llvm::AliasSetTracker *ast) {
+    if (!dyn_cast<StoreInst>(i) && !dyn_cast<LoadInst>(i)) return;
```

Comment:
LLVM's Instruction class has a method  [`mayReadOrWriteMemory `](https://llvm.org/doxygen/classllvm_1_1Instruction.html#a2d77b9d450543e86acb394ff6dda6b53), which we might want to consider using instead of this check. It considers some more esoteric instructions that I think we are unlikely to see in our benchmarks, but might be better to catch them correctly here (notably, I don't think it includes GEP instructions, so the gist of this strategy I think remains the same).

Reply:
Yes, that looks like what we want!